tests/test_cache.c

```diff
@@ -0,0 +1,38 @@
+
+void test_cache_invalidate(TestContext *ctx) {
+	uint8_t buf[128] __attribute__((aligned(16)));
+	const char *dfs_header = "\xde\xad\xbe\xef\xff\xff\xff\xff""DragonFS 2.0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00";
+	const char *aaa = "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa";
+
+	for (int i=0;i<32;i++) {
+		for (int j=0;j<32;j++) {
+
+			// Write the whole buffer through cache,
+			// so it's all populated in D-Cache.
+			memset(buf, 0xAA, sizeof(buf));
+
+			// Writeback+Invalidate buf[i..i+j]. Now only
+			// those lines should be invalidated.
+			data_cache_hit_writeback_invalidate(buf+i, j);
+			
+			// Read from ROM (header of DFS)
+			dma_read(UncachedAddr(buf), DFS_DEFAULT_LOCATION, 32);
```

Comment:
Does using an UncachedAddr make a difference here?

Reply:
Does it make sense to ever read from a cached address with DMA? I don't think so, maybe dma_read should auto-promote addresses to uncached for correctness? Not worth fixing in this PR however.

Reply:
yes it might be worth doing it within dma_read. I'll do it at some point I guess